src/util/bip32.rs

```diff
@@ -326,6 +326,12 @@ impl DerivationPath {
     pub fn hardened_children(&self) -> DerivationPathIterator {
         DerivationPathIterator::start_from(&self, ChildNumber::Hardened{ index: 0 })
     }
+
+    pub fn extend(&self, path: impl AsRef<[ChildNumber]>) -> DerivationPath {
+        let mut path = self.clone();
```

Comment:
you override the `path` input variable here
also, why does this create a new DerivationPath instead of extending the current? (I see that `child()` does the same)

and, why is `impl AsRef<[ChildNumber]>` better than a simple `&[ChildNumber]`?

Reply:
Number 3 makes this non-compiling for the current MSRV btw